### PR TITLE
WIP feature/Stepper

### PR DESCRIPTION
**Pull request by JMagic-Learner**
I have implemented a Stepper Component.

The current stepper component has 2 variations for horizontal display

### CODE REVIEW

File: src/components/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+                                btnType='primary'
+                                data-testid='button-element-prev'
+                                onClick={()=>prev(1)}
+                                btnSize={'sm'}
+                                disabled={Initialize}
+                                >
+                                {props.buttonTitlePrev}
+                        </Button>
+                        
+                        {skipButtonActive == true &&
```
Comment (seraphD) [inline src/components/Stepper/Stepper.tsx:261]: You can directly use boolean type variables like "skipButtonActive  &&  ..."

File: src/components/Stepper/Stepper.tsx [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+    const [skipButtonActive, setSkipButtonActive] = useState(false)
+    // LEGACY CODE FROM PREVIOUS VERSION OF STEPPER
+    // const [skipIndex, setSkipIndex] = useState<number>(9999999)
+    // const [SkipIndexArray, setSkipIndexArray] = useState<number[]>([])
+    const [Initialize, setInitialize] = useState(true)
+    let totalSteps = StepperElements!.length
+    let renderSteps = totalSteps - 1
+
+        useEffect(() => {
+        if (allowSkip == true) {
```
Comment (seraphD) [inline src/components/Stepper/Stepper.tsx]: This part is unnecessary. You can directly use the props.

File: src/components/Stepper/Stepper.tsx [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+            setSkipButtonActive(true)
+        }
+        if (StepperOrientation == 'vertical') {
+            setTriggerVertical("vertical");
+        }
+        
+        if (StepperOrientation == 'vertical') {
+            let currentTarget = 'description-area-' + `${Currentindex}`
+            let currentElement = document.getElementById(currentTarget)
```
Comment (seraphD) [inline src/components/Stepper/Stepper.tsx:69]: In React, I think we don't need to do things like "getElementById". You can directly add style props like "<div style={{ 
height: '20vh' }}>"

File: src/components/Stepper/Stepper.tsx [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+        }
+        
+        if (StepperOrientation == 'vertical') {
+            expand(direction,value)
+        }
+    }
+
+    let Stepper =
+        <div className="flex-container">
+        <div className={"center-main-body " + `${StepperOrientation}`} data-testid="center-main-body">
```
Comment (seraphD) [inline src/components/Stepper/Stepper.tsx:117]: Use the classnames generator under the utility folder like what u did with the size prop.

File: src/components/Stepper/Stepper.tsx (hunk 1 above)
Comment (seraphD) [inline src/components/Stepper/Stepper.tsx]: You can directly use boolean type variables like "skipButtonActive && ..."

File: src/components/Stepper/Stepper.tsx [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+                                    <p className={`font-variant-secondary `}> {item.description}</p>
+                                </div>
+                                }
+                            </div>
+                        </div> 
+                )
+             })}
+             </div>
+
+            {Currentindex == StepperElements?.length &&
```
Comment (seraphD) [inline src/components/Stepper/Stepper.tsx]: Use "===" instead of "==". Other places also.

File: src/components/Stepper/Stepper.tsx [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+                                onClick={()=>next(1)}
+                                btnSize={'sm'}
+                                >
+                                {props.buttonTitleNext}
+                        </Button>
+            </div>
+        </div>
+
+        {triggerVertical == 'vertical' &&
+            <div className="component-display" data-test-id="element-render-3">
```
Comment (seraphD) [inline src/components/Stepper/Stepper.tsx:261]: Bug check. When clicking the back button in the vertical stepper, animation happens on the previous step.

File: src/components/Stepper/_Stepper.scss [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+.Back {
+    width:fit-content;
+    margin:auto;
+}
+
+.spacer {
+    width:50%;
+}
+
+.sm {
```
Comment (seraphD) [inline src/components/Stepper/_Stepper.scss]: Define these constant size variables in styles/_variables.scss and use mixin functions to generate styles for different type of compoenents

File: src/components/Stepper/Stepper.test.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import Stepper, { patStepperProps } from './Stepper';
+import Button from '../Button'
+// import {renderer} from 'react-test-renderer'; // ES6
+
+describe ('Stepper', () => {
+
+// THE FOLLOWING 2 TESTS VERIFIES THE HTML STRUCTURE OF Stepper
```
Comment (seraphD) [inline src/components/Stepper/Stepper.test.tsx:12]: Need to add a snapshot test.

File: src/components/Stepper/Stepper.test.tsx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        expect(centermainbody).toHaveClass('center-main-body vertical');
+  
+        const rowelementRenderCheck2 = wrapper.getByTestId("element-render-2")
+        const rowelementRenderCheck3 = wrapper.getByTestId("element-render-3")
+       
+        expect(rowelementRenderCheck2).toBeInTheDocument();
+        expect(rowelementRenderCheck2).toHaveClass('vertical');
+        expect(rowelementRenderCheck3).toBeInTheDocument();
+        expect(rowelementRenderCheck3).toHaveClass('component-display');
+    })
```
Comment (seraphD) [inline src/components/Stepper/Stepper.test.tsx:104]: I recommend adding some more tests on props and some edge cases.

File: src/components/Stepper/Stepper.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,465 @@
...
+  marginTop: '5px',
+};
+
+export default {
+    title: 'Stepper',
+    component: Stepper
+}
+
+
+export const DefaultStepper = () => (
```
Comment (seraphD) [inline src/components/Stepper/Stepper.stories.tsx:18]: Default Stepper should not include any customized components. Show them in another story.

**Review by seraphD — COMMENTED** [review]
Good work with the Stepper compoenent. I can see you put lots of effort into it. My suggestion is just to make your code more professional by improving some details as I commented.

File: src/components/Stepper/Stepper.tsx (hunk 1 above)
Comment (JMagic-Learner) [reply src/components/Stepper/Stepper.tsx]: Received and changed

File: src/components/Stepper/Stepper.tsx (hunk 3 above)
Comment (JMagic-Learner) [reply src/components/Stepper/Stepper.tsx:69]: Followed your advice on using props to dictate height expansion for Vertical component



` <div className={"description-area " + StepperSize } style={{height: ExpandBy }} id={"description-area-" + index } data-testid={`description-area-` + `${index}`}>`

Question: Changing props or changing the value of the style props affects the height of all mapped containers className={"description-area " + StepperSize }

I am assuming that there is a way narrow the state / props to only affect a single mapped container.

For example. the current description-area container should be the only one that is affected.

Comment (seraphD) [reply src/components/Stepper/Stepper.tsx:69]: Great

File: src/components/Stepper/_Stepper.scss (hunk 7 above)
Comment (JMagic-Learner) [reply src/components/Stepper/_Stepper.scss]: Left sm and lg so user can pass down props into Stepper component. Also added StepperSize props to Button. Added some mixins and styling to _Stepper.scss

File: src/components/Stepper/Stepper.tsx (hunk 6 above)
Comment (JMagic-Learner) [reply src/components/Stepper/Stepper.tsx:261]: I think i resolved this with `style={{height: `${index === Currentindex ? expandHeight : initialHeight}` }}`
